packages/features/webhooks/lib/factory/BookingPayloadBuilder.ts

```diff
@@ -0,0 +1,206 @@
+import { getUTCOffsetByTimezone } from "@calcom/lib/dayjs";
+import { WebhookTriggerEvents } from "@calcom/prisma/enums";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+
+import type { EventTypeInfo, BookingWebhookEventDTO } from "../dto/types";
+import type { WebhookPayload } from "./types";
+
+type BookingExtraDataMap = {
+  [WebhookTriggerEvents.BOOKING_CREATED]: undefined;
+  [WebhookTriggerEvents.BOOKING_CANCELLED]: { cancelledBy?: string; cancellationReason?: string };
+  [WebhookTriggerEvents.BOOKING_REQUESTED]: undefined;
+  [WebhookTriggerEvents.BOOKING_REJECTED]: undefined;
+  [WebhookTriggerEvents.BOOKING_RESCHEDULED]: {
+    rescheduleId?: number;
+    rescheduleUid?: string;
+    rescheduleStartTime?: string;
+    rescheduleEndTime?: string;
+    rescheduledBy?: string;
+  };
+  [WebhookTriggerEvents.BOOKING_PAID]: { paymentId?: number; paymentData?: Record<string, unknown> };
+  [WebhookTriggerEvents.BOOKING_PAYMENT_INITIATED]: {
+    paymentId?: number;
+    paymentData?: Record<string, unknown>;
+  };
+};
+
+function createBookingWebhookPayload<T extends keyof BookingExtraDataMap>(
+  booking: {
+    id: number;
+    eventTypeId: number | null;
+    userId: number | null;
+    smsReminderNumber?: string | null;
+  },
+  eventType: EventTypeInfo,
+  evt: CalendarEvent,
+  status: string,
+  triggerEvent: string,
```

Comment:
Looks like triggerEvent should be an enum as well.

Reply:
It should be, Will do